django_app/redbox_app/redbox_core/models.py

```diff
@@ -481,3 +481,19 @@ def get_messages_ordered_by_citation_priority(cls, chat_id: uuid.UUID) -> Sequen
                 )
             )
         )
+
+
+class ChatMessageTokenUse(UUIDPrimaryKeyBase, TimeStampedModel):
+    class UseTypeEnum(models.TextChoices):
+        INPUT = "input", _("input")
+        OUTPUT = "output", _("output")
+
+    chat_message = models.ForeignKey(ChatMessage, on_delete=models.CASCADE)
+    use_type = models.CharField(
+        max_length=10, choices=UseTypeEnum, help_text="input or output tokens", default=UseTypeEnum.INPUT
+    )
+    model_name = models.CharField(max_length=50, null=True, blank=True)
```

Comment:
in principle i think this should be https://github.com/i-dot-ai/redbox/blob/fb949dbe5b67b64be69955c6e0a52740b9227525/django_app/redbox_app/redbox_core/models.py#L49 but lets not worry about that now